returnn/training.py

```diff
@@ -159,13 +180,21 @@ def __init__(
             for k in stored_epochs
             if k in self.keep_epochs
         }
-        if self.returnn_config.get("use_tensorflow", False):
+        if self.returnn_config.get("use_tensorflow", False) or self.returnn_config.get("backend") == "tensorflow":
             self.out_checkpoints = {
                 k: Checkpoint(index_path)
                 for k in stored_epochs
                 if k in self.keep_epochs
                 for index_path in [self.output_path("models/epoch.%.3d.index" % k)]
             }
+        if self.returnn_config.get("backend", None) == "torch":
+            self.out_checkpoints = {
+                k: PtCheckpoint(index_path)
+                for k in stored_epochs
+                if k in self.keep_epochs
+                for index_path in [self.output_path("models/epoch.%.3d.pt" % k)]
+            }
```

Comment:
Since this is almost purely code duplication, couldn't this be unified into something like
```suggestion
        if self.returnn_config.get("use_tensorflow", False) or self.returnn_config.get("backend") == "tensorflow":
            checkpoint_class = Checkpoint
            suffix = "index"
        if self.returnn_config.get("backend", None) == "torch":
            checkpoint_class = PtCheckpoint
            suffix = "pt"

        self.out_checkpoints = {
            k: checkpoint_class(index_path)
            for k in stored_epochs
            if k in self.keep_epochs
            for index_path in [self.output_path(f"models/epoch.{k:03d}.{suffix}")]
        }
```
?